### ISSUE TITLE
cabal (v2-)install --lib lib:name fails with non-exhaustive pattens in Just ghcPkgProg

### ISSUE THREAD

**Issue by maoe** [haskell/cabal#5941]
I was trying to install a library from source and `cabal install --lib lib:NAME` failed.

Steps to reproduce with cabal-install from recent master (e23d8d1):
```console
% git clone git@github.com:bgamari/ghc-dump.git
% cd ghc-dump
% cabal install --lib lib:ghc-dump-core
Wrote tarball sdist to
/Users/maoe/src/github.com/bgamari/ghc-dump/dist-newstyle/sdist/ghc-dump-core-0.1.0.0.tar.gz
Wrote tarball sdist to
/Users/maoe/src/github.com/bgamari/ghc-dump/dist-newstyle/sdist/ghc-dump-util-0.1.0.0.tar.gz
Resolving dependencies...
Up to date
Distribution/Simple/GHC.hs:1940:5-70: Non-exhaustive patterns in Just ghcPkgProg
```

Apparently `progdb` in `hcPkgInfo` is empty when the command fails.

**Comment by maoe** [comment]
FWIW if I don't pass the `--lib` flag, it doesn't get this error:

```
% cabal install lib:ghc-dump-core
Wrote tarball sdist to
/Users/maoe/src/github.com/bgamari/ghc-dump/dist-newstyle/sdist/ghc-dump-core-0.1.0.0.tar.gz
Wrote tarball sdist to
/Users/maoe/src/github.com/bgamari/ghc-dump/dist-newstyle/sdist/ghc-dump-util-0.1.0.0.tar.gz
Resolving dependencies...
Up to date
Warning: You asked to install executables, but there are no executables in
target: ghc-dump-core. Perhaps you want to use --lib to install libraries
instead.
```

**Comment by hvr** [comment]
I believe this was fixed by #6195 (which was also included in the 3.0.0.0 release)

**Comment by maoe** [comment]
Thanks for letting me know. I confirmed that `cabal install --lib lib:ghc-dump-core` worked fine with cabal-install 3.0.0.0. Closing.
